app/middlewares/album.js

```
@@ -1,18 +1,41 @@
 const { albumUser } = require('../models'),
-  { alreadyBoughtAlbumError } = require('../errors');
+  { alreadyBoughtAlbumError, didntBuyAlbumError } = require('../errors'),
+  logger = require('../logger');
 
-exports.checkIfUserBoughtAlbum = (req, res, next) =>
-  albumUser
+const userBoughtAlbum = (req, res, next) => {
+  logger.info('Searching album...');
+  return albumUser
     .findOne({
       where: {
         albumId: req.params.id,
         userId: req.user.id
       }
     })
-    .then(album => {
-      if (album) {
+    .then(album => album)
+    .catch(next);
```

you can throw an error here